### DISCUSSION TITLE
HomeAssistent cannot connect to ebusd

### DISCUSSION THREAD

**Discussion by koenigfb** [john30/ebusd#1364] [Q&A] [answered]
**Issue**

I managed to set up Adapter Shield v5 via WiFi to MQTT. Messages are incoming from the heater:

```
2024-10-22 12:51:47.673 [bus debug] ERR: SYN received during receive command ACK, switching to ready
2024-10-22 12:51:47.800 [bus debug] ERR: SYN received during receive command ACK, switching to ready
2024-10-22 12:51:47.925 [bus debug] ERR: SYN received during receive command ACK, switching to ready
2024-10-22 12:51:48.050 [bus debug] ERR: SYN received during receive command ACK, switching to ready
2024-10-22 12:51:48.288 [mqtt debug] publish ebusd/feuerung/betrd {
     "status": {"value": "Brauchwasser_Heizen"},
     "aktion": {"value": "Verbraucheraus"},
     "kesselsolltemp": {"value": 29.38},
     "solldruck": {"value": null},
     "stellgrad": {"value": null},
     "brauchwassersolltemp": {"value": 50.0},
     "brennstoff": {"value": "dc"}}
2024-10-22 12:51:48.288 [mqtt debug] publish ebusd/broadcast/betrd {
     "status": {"value": 0},
     "zustand": {"value": "48"},
     "stellgrad": {"value": null},
     "kesseltemp": {"value": 39.5},
     "ruecklauftemp": {"value": 29},
     "boilertemp": {"value": 45},
     "aussentemp": {"value": 15}}

```

However Autodiscovery of the sensors isnt working. I am getting this error during the HA restart:

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.12/site-packages/ebusdpy/ebusdpy.py", line 14, in init
    sock.connect(address)
ConnectionRefusedError: [Errno 111] Connection refused
```

I tried different configurations, but none worked:

```
ebusd:
  host: 127.0.0.1  //also tried the HA IP
  port: 9999  //tried 8888
  circuit: "bai" //tried others as well e.g. feuerung

```

This is the ebusd config:

```
scanconfig: true
loglevel_all: debug
mqtttopic: ebusd
mqttint: /etc/ebusd/mqtt-hassio.cfg
mqttjson: true
mode: ens
network_device: 192.168.178.77:9999
latency: 10
configpath: /config/wolf
readonly: false
http: true
commandline_options: "--configlang=DE"  #set it because csv is in german
```

Does anyone know what might be wrong here?

ebusd version
23.2.6

ebusd arguments
--configlang=DE

Operating system
Homeassistent

Hardware interface
Adapter Shield v5 via WiFi

Related integration
MQTT Home Assistant

**Comment by Koky05** [comment, marked as answer]
How do you run ebusd? I assume as HA Addon, so do you create local folder with mqtt-hassio.cfg?
It looks like your HA try to connect directly to ebus but it should communicate via MQTT to ebusd.

More detail how to configure it:
[Home Assistant Add-on: eBUSd](https://github.com/LukasGrebe/ha-addons/blob/main/ebusd/DOCS.md)

**Comment by koenigfb** [comment]
Yes, I am using the HA Addon.

I now added the mqtt-hassio.cfg file (https://github.com/john30/ebusd/blob/master/contrib/etc/ebusd/mqtt-hassio.cfg) to the /config folder without any adjustments as I understand that mqtt connection is done by autodiscovery if I dont override in the options.

If I am seeing those sensor here, does it mean its already integrated? Its just not picking up the other heating sensors from mqtt?
![grafik](https://github.com/user-attachments/assets/89a121b3-d51d-4baf-8734-a64f30340e6c)


If HA is communicating via MQTT why do I need to define this in HA configuration file?

```
ebusd:
  host: 127.0.0.1  //also tried the HA IP
  port: 9999  //tried 8888
  circuit: "bai" //tried others as well e.g. feuerung
```

**Reply by koenigfb** [reply]
ok, it worked after 
- adding the mqtt-hassio.cfg
- adjusting the filter
(- removing ebusd entry HA configuration file)

Thank you!
